salt/master.py

```diff
@@ -2121,24 +2144,27 @@ def publish(self, clear_load):
                         'error': 'Master could not resolve minions for target {0}'.format(clear_load['tgt'])
                     }
                 }
-        jid = self._prep_jid(clear_load, extra)
-        if jid is None:
-            return {'enc': 'clear',
-                    'load': {'error': 'Master failed to assign jid'}}
-        payload = self._prep_pub(minions, jid, clear_load, extra, missing)
-
-        # Send it!
-        self._send_ssh_pub(payload, ssh_minions=ssh_minions)
-        self._send_pub(payload)
-
-        return {
-            'enc': 'clear',
-            'load': {
-                'jid': clear_load['jid'],
-                'minions': minions,
-                'missing': missing
+        if batch:
+            return self.publish_batch(batch, clear_load)
+        else:
```

Comment:
I guess the `else:` clause is not really necessary here since we `return`, and this would make the patch smaller/easier to see we are not changing the behaviour of this method unless the new parameter is set.